src/WebJobs.Script/Config/FunctionsHostingConfigurations.cs

```diff
@@ -0,0 +1,100 @@
+        {
+            _environment = environment;
+            _logger = loggerFactory.CreateLogger<FunctionsHostingConfigurations>() ?? throw new ArgumentNullException(nameof(loggerFactory));
+            _configFilePath = environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsRuntimeConfigFilePath);
+            if (string.IsNullOrEmpty(_configFilePath))
+            {
+                _configFilePath = Environment.ExpandEnvironmentVariables(Path.Combine("%ProgramFiles(x86)%", "SiteExtensions", "kudu", "ScmHostingConfigurations.txt"));
+            }
+        }
+
+        // For tests
+        internal FunctionsHostingConfigurations (IEnvironment environment, ILoggerFactory loggerFactory, string configsFile, DateTime configsTTL, TimeSpan updateInterval)
+            : this(environment, loggerFactory)
+        {
+            _configFilePath = configsFile;
+            _configTTL = configsTTL;
+            _updateInterval = updateInterval;
+        }
+
+        // for tests
+        internal Dictionary<string, string> Config => _configs;
+
+        public bool FunctionsWorkerDynamicConcurrencyEnabled
+        {
+            get
+            {
+                // Worker concurrecy feature can be enabled for whole stamp or for individual apps.
+                string value = GetValue(RpcWorkerConstants.FunctionsWorkerDynamicConcurrencyEnabled, null);
+                if (!string.IsNullOrEmpty(value))
+                {
+                    var values = value.Split(";").Select(x => x.ToLower());
+                    string siteName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteName);
+                    return values.Contains("stamp") || values.Contains(siteName);
+                }
+                return false;
+            }
+        }
+
+        public string GetValue(string key, string defaultValue = null)
+        {
+            var configs = _configs;
+            if (configs == null || DateTime.UtcNow > _configTTL)
+            {
+                lock (_configFilePath)
+                {
+                    _configTTL = DateTime.UtcNow.Add(_updateInterval);
+
+                    if (FileUtility.FileExists(_configFilePath))
+                    {
+                        var settings = FileUtility.ReadAllText(_configFilePath);
+                        configs = Parse(settings);
+                        _configs = configs;
+                        _logger.LogInformation($"Updaiting FunctionsHostingConfigurations '{settings}'");
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException("FunctionsHostingConfigurations does not exists");
+                    }
+                }
+            }
+
+            return (configs == null || !configs.TryGetValue(key, out string value)) ? defaultValue : value;
+        }
+
+        private static Dictionary<string, string> Parse(string settings)
```

Comment:
Can we add a comment on wha the setting might look like that is getting parsed here?